include/cuda/std/detail/libcxx/include/array

```diff
@@ -483,4 +483,25 @@ get(const array<_Tp, _Size>&& __a) _NOEXCEPT
 
 _LIBCUDACXX_END_NAMESPACE_STD
 
+// This is a workaround for the fact that structured bindings require that the specializations of
+// `tuple_size` and `tuple_element` reside in namespace std (https://eel.is/c++draft/dcl.struct.bind#4).
+// See https://github.com/NVIDIA/libcudacxx/issues/316 for a short discussion
+namespace std {
+    template<class>
+    struct tuple_size;
+    
+    template<size_t, class>
+    struct tuple_element;
+
+    template <class _Tp, size_t _Size>
+    struct tuple_size<_CUDA_VSTD::array<_Tp, _Size>> 
+      : _CUDA_VSTD::tuple_size<_CUDA_VSTD::array<_Tp, _Size>> 
+    {};
```

Comment:
libcxx is failing to build due to this:

```
2022-09-22 06:23:19+00:00| In file included from /sw/gpgpu/libcudacxx/libcxx/src/filesystem/filesystem_common.h:14:
2022-09-22 06:23:19+00:00| /sw/gpgpu/libcudacxx/libcxx/include/array:497:12: error: explicit specialization of non-template struct 'tuple_size'
2022-09-22 06:23:19+00:00| struct tuple_size<_CUDA_VSTD::array<_Tp, _Size>>
2022-09-22 06:23:19+00:00| ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-22 06:23:19+00:00| /sw/gpgpu/libcudacxx/libcxx/include/array:502:12: error: explicit specialization of non-template struct 'tuple_element'
2022-09-22 06:23:19+00:00| struct tuple_element<_Ip, _CUDA_VSTD::array<_Tp, _Size>>
2022-09-22 06:23:19+00:00| ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-22 06:23:19+00:00| In file included from /sw/gpgpu/libcudacxx/libcxx/src/filesystem/operations.cpp:10:
2022-09-22 06:23:19+00:00| /sw/gpgpu/libcudacxx/libcxx/include/array:497:12: error: explicit specialization of non-template struct 'tuple_size'
2022-09-22 06:23:19+00:00| struct tuple_size<_CUDA_VSTD::array<_Tp, _Size>>
2022-09-22 06:23:19+00:00| ^         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-22 06:23:19+00:00| /sw/gpgpu/libcudacxx/libcxx/include/array:502:12: error: explicit specialization of non-template struct 'tuple_element'
2022-09-22 06:23:19+00:00| struct tuple_element<_Ip, _CUDA_VSTD::array<_Tp, _Size>>
2022-09-22 06:23:19+00:00| ^            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2022-09-22 06:23:19+00:00| 1 warning and 2 errors generated.
```

Reply:
But the forward declaration is like directly above. How can that fail?